processing/src/main/java/org/apache/druid/query/QueryContexts.java

```diff
@@ -604,18 +644,31 @@ private QueryContexts()
   {
   }
 
-  static <T, E extends Enum<E>> E parseEnum(Query<T> query, String key, Class<E> clazz, E defaultValue)
```

Comment:
this class is also marked `@PublicApi` and so we should ideally leave the signatures in place

Reply:
But this method is not declared as public. Do we need to keep it?